src/Microsoft.OData.Edm/Schema/EdmNavigationSource.cs

```diff
@@ -110,8 +114,10 @@ public virtual IEnumerable<IEdmNavigationPropertyBinding> FindNavigationProperty
         {
             EdmUtil.CheckArgumentNull(navigationProperty, "navigationProperty");
 
-            Dictionary<string, IEdmNavigationPropertyBinding> result;
-            if (this.navigationPropertyMappings.TryGetValue(navigationProperty, out result))
+            Dictionary<string, IEdmNavigationPropertyBinding> result =
+                EdmUtil.DictionarySafeGet(this.navigationPropertyMappings, navigationProperty);
+
+            if (result != null)
             {
                 return result.Select(item => item.Value);

```

Comment:
>result.Select(item => item.Value); [](start = 23, length = 34)

why can't return result.Values?

Reply:
I didn't change anything here, but it does seem like we could just return result.Values.